### PR TITLE
Add explicit include for src/**/docs/*.md files in .npmignore to fix `!raw-loader!` imports

### PR DESCRIPTION
**Pull request by jordandrako**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

When importing data directly from one of the .doc.ts files when OUFR is not local (just a package dependency), webpack cannot find the files since the `src` folder isn't published in the NPM package. This is blocking a change we're trying to make to the new HIG site.

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/7887)

### CODE REVIEW

**Comment by jordandrako** [comment]
So the build is failing due to the markdown files not being copied over to the lib folder on build.

**Comment by kenotron** [comment]
There was a reason why these were src before :) the normal "pre-copy.json" thing might not work ...

**Comment by jordandrako** [comment]
@kenotron I'll keep experimenting... Is there a way for me to have this travis test run without this pull request? I don't want to keep it open with a whole bunch of changes we may not merge.

**Comment by Jahnp** [comment]
Hey @jordandrako , I think all we'd need to do here is publish the `docs` folders in the npm package by adding `!src/**/docs/*.md` to `packages/office-ui-fabric-react/.npmignore`. We're already doing that with the *.types.ts and example files--I think this should work for our scenario. I'll try testing locally. Any concerns with/thoughts about this @kenotron ?

Commit here for reference: https://github.com/Jahnp/office-ui-fabric-react/commit/9a910dfd20d8a935f9f52fcf674df88655eaac4a

**Comment by Jahnp** [comment]
Verified locally that this should work by manually copying `docs` some folders into a local `node_modules` installation. We can proceed with this tomorrow if there're no concerns.

File: packages/office-ui-fabric-react/src/components/DetailsList/__snapshots__/DetailsHeader.test.tsx.snap [hunk 1]
```diff
@@ -889,6 +889,7 @@ exports[`DetailsHeader can render a hidden select all checkbox in single selecti
         line-height: 32px;
         min-width: 100%;
         padding-bottom: 1px;
+        padding-top: 1px;
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/DetailsList/__snapshots__/DetailsHeader.test.tsx.snap]: Was this supposed to be part of this change?

Comment (jordandrako) [reply packages/office-ui-fabric-react/src/components/DetailsList/__snapshots__/DetailsHeader.test.tsx.snap]: I had screener failing before running update-snapshots.

**Comment by KevinTCoughlin** [comment]
> Just a note that this will increase the size of the npm package (see #7255), though it's needed to consume the .md docs outside of this repo. If that's a problem, we could consider moving those to a separate package, e.g. something like @uifabric/react-docs.

The title of this PR is a bit misleading since this is adding an explicit include for these files to our `npm pack` output. I would advocate for moving them to a separate package rather than inflate our library distributable. Markdown / documentation files can grow unbounded and conflict with #7255.

How much work would it be to consume them as part of a separate package?

I am in favor of fixing the issue, but hope that we can do so without inflating tarball size.

**Comment by KevinTCoughlin** [comment]
Do people that `npm install office-ui-fabric-react` expect to be able to use the markdown files?

**Comment by Jahnp** [comment]
@KevinTCoughlin we are doing that for the HIG site, though we're also a "special case" since we're on the Fabric team itself 😆Ultimately though I totally agree with your concern, which is why I wanted to bubble it up here. We're already doing it with examples, though, so we kinda did start climbing that slippery slope.

It might help to explain the motivation for this change. For some time, the HIG website has basically been an internal fork of OUIFR, with all the goodness and grossness that comes along with maintaining a fork. It's meant to be a version of @uifabric/fabric-website that builds on the OSS .md files and adds additional content for MS employees. During that time, the HIG could easily consume these .md files since they were in a peer package (i.e. there is *literally* a `packages/office-ui-fabric-react` that mirrors upstream that we could read from). 

But it turns out constantly running `git merge -X theirs upstream/master`, dealing with conflicts and codebase-wide changes, long build times, etc. is painful and hurts productivity, so we're working on improving the DevExp by cleaning out the public packages from the fork. That's what led to this issue—we basically can't consume these .md files since they aren't published anywhere.

I'm definitely open to moving them to a separate package, but that'd probably require moving all the `components/**/docs/**/*.md` files, which would separate them from their actual components. Or maybe doing some weird pre-release scripting to copy the .md into a package.

Thoughts?

File: packages/office-ui-fabric-react/src/components/DetailsList/__snapshots__/DetailsHeader.test.tsx.snap (hunk 1 above)
Comment (jordandrako) [reply packages/office-ui-fabric-react/src/components/DetailsList/__snapshots__/DetailsHeader.test.tsx.snap]: @ecraig12345 Merging master fixed this.

**Comment by KevinTCoughlin** [comment]
I defer to @kenotron :)

**Pull request merged**